sdk/jobs/pipelines/1h_automl_in_pipeline/automl-classification-bankmarketing-in-pipeline/register.py

```diff
@@ -0,0 +1,94 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import argparse
+import json
+import os
+import time
+
+
+from azureml.core import Run
+
+import mlflow
+import mlflow.sklearn
+
+# Based on example:
+# https://docs.microsoft.com/en-us/azure/machine-learning/how-to-train-cli
+# which references
+# https://github.com/Azure/azureml-examples/tree/main/cli/jobs/train/lightgbm/iris
+
+
+def parse_args():
+    # setup arg parser
+    parser = argparse.ArgumentParser()
+
+    # add arguments
+    parser.add_argument("--model_input_path", type=str, help="Path to input model")
+    parser.add_argument(
+        "--model_info_output_path", type=str, help="Path to write model info JSON"
+    )
+    parser.add_argument(
+        "--model_base_name", type=str, help="Name of the registered model"
+    )
+    parser.add_argument(
+        "--model_name_suffix", type=int, help="Set negative to use epoch_secs"
+    )
+
+    # parse args
+    args = parser.parse_args()
+
+    # return args
+    return args
+
+
+def main(args):
+    current_experiment = Run.get_context().experiment
+    tracking_uri = current_experiment.workspace.get_mlflow_tracking_uri()
+    print("tracking_uri: {0}".format(tracking_uri))
+    mlflow.set_tracking_uri(tracking_uri)
+    mlflow.set_experiment(current_experiment.name)
+
+    print("Getting model path")
+    mlmodel_path = os.path.join(args.model_input_path, "MLmodel")
+    runid = ""
+    with open(mlmodel_path, "r") as modelfile:
+        while(True):
+            line = modelfile.readline()
+            if not line:
+                break
+            if "run_id" in line:
+                runid = line.split(":")[1].strip()
+    model_uri = "runs:/{}/outputs/".format(runid)
+    print("Model URI: " + model_uri)
+
+    if args.model_name_suffix < 0:
+        suffix = int(time.time())
+    else:
+        suffix = args.model_name_suffix
+    registered_name = "{0}_{1}".format(args.model_base_name, suffix)
+    print(f"Registering model as {registered_name}")
+    mlflow.register_model(model_uri, registered_name)
```

Comment:
Is mlflow the recommended way? 
or can we make use of mlclient 

```
model = load_model(path=file)
model = ml_client.create_or_update(model)
```

Reply:
We can use ml_client too, but since mlfow is already used to set tracking uri, keeping the changes as is.